ws-nextjs-app/utilities/cspHeaderResponse/index.ts

```diff
@@ -0,0 +1,72 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { helmetCsp } from '#server/utilities/cspHeader/directives';
+import { Services } from '#app/models/types/global';
+
+const directiveToString = (directives: Record<string, string | string[]>) => {
+  const map = new Map(Object.entries(directives));
+  let cspValue = '';
+  map.forEach((allowList, directive) => {
+    const allowListString = [allowList]
+      .flat()
+      .reduce((curr, acc) => `${curr} ${acc}`, '')
+      .replace(/\s{2,}/g, ' ');
+    cspValue += `${directive}${allowListString.length > 0 ? ' ' : ''}${allowListString};`;
+  });
+  return cspValue;
+};
+
+const cspHeaderResponse = ({
+  request,
+  isAmp = false,
+  isLive = false,
+  reportOnlyOnLive = false,
+  service,
+}: {
+  request: NextRequest;
+  isAmp?: boolean;
+  isLive?: boolean;
+  reportOnlyOnLive?: boolean;
+  service: Services;
+}) => {
+  const requestHeaders = new Headers(request.headers);
+
+  const helmetCSP = helmetCsp({
+    isAmp,
+    isLive,
+    reportOnlyOnLive,
+    service,
+  });
```

Comment:
Could this be renamed to something more generic than `helmetCsp`? HelmetCsp is an actual library, so just don't want to cause confusion that the Next.js app will actually be using that library.